memgpt/client/client.py

```diff
@@ -0,0 +1,74 @@
+from typing import Dict, List, Union
+
+from memgpt.config import AgentConfig
+from memgpt.persistence_manager import PersistenceManager
+from memgpt.server.rest_api.interface import QueuingInterface
+from memgpt.server.server import SyncServer
+
+
+class Client(object):
+    def __init__(
+        self,
+        # Utility settings:
+        auto_save: bool = False,
+        # Allow running quickstart on a client init
```

Comment:
@BabellDev what do you think about this extra args? Basically I'm thinking it should be possible to do something like the following without even having run `memgpt configure` on a fresh install:

```python
from memgpt import MemGPT

client = MemGPT(
  auto_save=True,
  quickstart="memgpt"
)
```

or
client = MemGPT(
  auto_save=True,
  quickstart="openai",
  config={"openai_api_key": "sk-..."},
)
```

The exact default values / types aren't finished they're just easy placeholders (should probably be unions of pydantic models that are defined elsewhere for server).

Reply:
Yeah this is a great idea.  If we can avoid calling `memgpt configure`, then it's even easier to get started with this.

Reply:
Make some adjustments and tested with completely empty ~/.memgpt dir and it seems to work